src/components/PassageDataTable.tsx

```diff
@@ -165,32 +180,28 @@ export const SelectSharedResource = (props: IProps) => {
     onOpen && onOpen(false);
   };
 
-  const numSort = (i: number, j: number) => i - j;
-
   const handleCheck = (chks: Array<number>) => {
-    const curLen = checks.length;
-    const newLen = chks.length;
-    if (curLen < newLen) {
-      const termsList: number[] = [];
-      const noTermsList: number[] = [];
-      for (const c of chks.sort(numSort)) {
-        if (!checks.includes(c)) {
-          if (termsOfUse(c)) {
-            termsList.push(c);
-          } else {
-            noTermsList.push(c);
-          }
+    //we want single select so if there are more than one, we take the last one
+    if (chks.length > 1) chks = [chks[chks.length - 1]];
+    const termsList: number[] = [];
+    const noTermsList: number[] = [];
+    for (const c of chks) {
+      //.sort(numSort)) {
+      if (!checks.includes(c)) {
+        if (termsOfUse(c)) {
+          termsList.push(c);
+        } else {
+          noTermsList.push(c);
         }
       }
-      if (noTermsList.length > 0) {
-        setChecks(checks.concat(noTermsList).sort(numSort));
-      }
-      if (termsList.length > 0) {
-        setTermsCheck(termsList);
-        setCurTermsCheck(termsList[0]);
-      }
-    } else if (curLen > newLen) {
-      setChecks(chks);
+    }
+    //
+    if (noTermsList.length > 0) {
+      setChecks(noTermsList);
+    }
+    if (termsList.length > 0) {
+      setTermsCheck(termsList);
+      setCurTermsCheck(termsList[0]);
     }
   };
```

Comment:
This logic was primarily used when the shared resources are using this control. It seems like this change will not work if the user wants to include multiple shared resources?